app/components/access-level-pill.js

```diff
@@ -58,11 +56,13 @@ export default class AccessLevelPillComponent extends Component {
     return this.intl.t('no-accessLevel');
   }
 
+  get canEdit() {
+    return this.session.may('manage-document-access-levels');
+  }
+
   @action
   toggleEdit() {
-    if (this.session.isEditor) {
-      this.isEditing = !this.isEditing;
-    }
+    this.isEditing = !this.isEditing;
```

Comment:
Since this action is only used to open the edit mode I have a preference of naming it `openEditMode` and just doing `this.isEditing = true`. Edit mode is disabled as part of other actions; never through this action.